src/electionguard-ui/ElectionGuard.UI/ViewModels/CreateMultiTallyViewModel.cs

```diff
@@ -0,0 +1,353 @@
+                KeyCeremonies.Add(item);
+            }
+        }
+    }
+
+    [RelayCommand]
+    private void SelectionChanged()
+    {
+        if(SelectedElections.Count != Elections.Count)
+        {
+            SelectAll = false;
+        }
+        CreateTalliesCommand.NotifyCanExecuteChanged();
+        JoinTalliesCommand.NotifyCanExecuteChanged();
+    }
+
+    private async Task<string> CreateNewTally(ElectionItem election, bool multi=false)
+    {
+        ElectionGuardException.ThrowIfNull(election, $"Could not load election selected");
+        ElectionGuardException.ThrowIfNull(election.Election, $"ElectionItem does not contain an election");
+        ElectionGuardException.ThrowIfNull(election.Election.ElectionId, $"Election does not have an election id");
+        ElectionGuardException.ThrowIfNull(election.Election.KeyCeremonyId, $"Election does not have a key ceremony id");
+
+        // calculate ballot count and upload count
+        var ballotCount = await _ballotService.GetCountBallotsByElectionIdStateAsync(election.Election.ElectionId, BallotBoxState.Cast);
+        var challengedCount = await _ballotService.GetCountBallotsByElectionIdStateAsync(election.Election.ElectionId, BallotBoxState.Challenged);
+        var spoiledCount = await _ballotService.GetCountBallotsByElectionIdStateAsync(election.Election.ElectionId, BallotBoxState.Spoiled);
+
+        var keyCeremony = await _keyCeremonyService.GetByKeyCeremonyIdAsync(election.Election.KeyCeremonyId);
+        ElectionGuardException.ThrowIfNull(keyCeremony, $"Could not load key ceremony {election.Election.KeyCeremonyId}");
+
+        var tallyName = $"{election.Election.Name} {AppResources.TallyText}";
+
+        TallyRecord newTally = new()
+        {
+            Name = tallyName,
+            ElectionId = election.Election.ElectionId,
+            KeyCeremonyId = election.Election.KeyCeremonyId,
+            Quorum = keyCeremony.Quorum,
+            NumberOfGuardians = keyCeremony.NumberOfGuardians,
+            CastBallotCount = ballotCount,
+            ChallengedBallotCount = challengedCount,
+            SpoiledBallotCount = spoiledCount,
+            State = TallyState.PendingGuardiansJoin,
+            MultiTally = multi,
+        };
+
+        _ = await _tallyService.SaveAsync(newTally);
+
+        return newTally.TallyId;
+    }
+
+
+    [RelayCommand(CanExecute = nameof(TalliesSelected))]
+    private async Task CreateTallies()
+    {
+        if(SelectedElections.Count == 1)
+        {
+            // do a normal tally for a single election
+            var election = SelectedElections.First() as ElectionItem;
+
+            var tallyId = await CreateNewTally(election);
+
+            // go to the processing page
+            await NavigationService.GoToPage(typeof(TallyProcessViewModel), new Dictionary<string, object>
+            {
+                { TallyProcessViewModel.CurrentTallyIdParam, tallyId }
+            });
+        }
+        else

```

Comment:
This is a smaller example... There's definitely a place, but this pattern adds cyclomatic complexity, distinct from MSIL complexity or efficiency